src/iosMain/kotlin/com/liftric/job/queue/NetworkListener.kt

```diff
@@ -0,0 +1,40 @@
+package com.liftric.job.queue
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+
+actual class NetworkListener(
+    networkManager: NetworkManager,
+    scope: CoroutineScope = CoroutineScope(context = Dispatchers.Default)
+) : AbstractNetworkListener(
+    networkManager = networkManager,
+    scope = scope
+) {
+    private val _currentNetworkState = MutableStateFlow(NetworkState.NONE)
+    override val currentNetworkState: StateFlow<NetworkState>
+        get() = _currentNetworkState.asStateFlow()
+
+    override fun observeNetworkState() {
+        networkManager.startMonitoring()
+        scope.launch {
+            networkManager.network.collectLatest { currentNetworkState ->
+                _currentNetworkState.emit(
```

Comment:
If I did not misunderstand your implementation, you could also dismiss the `when` block and directly write:

```kotlin
_currentNetworkState.emit(currentNetworkState)
```

Reply:
Small note: in this case you could simplify it with (same applies to the android side):

```kotlin
_currentNetworkState.emit(it)
```